### DISCUSSION TITLE
User Gif and video presets

### DISCUSSION THREAD

**Discussion by andybak** [icosa-gallery/open-brush#194] [Ideas and Feature Requests]
For gifs there is a list of presets configured in the main scene:

![image](https://user-images.githubusercontent.com/60035/140733140-4c12d515-0a0a-455b-9cb2-8db0c8e1775c.png)

(on the MultiCamTool)

I propose we also read a list of additional presets from the config file so users can add their own.

It might also be useful to add "quality" to the preset option for gifs. We currently use the default quality of 10 from uGif.

**Comment by Gemcrafter** [comment]
Function notes: will add as I go

Frame Delay:
frame delay seems to be "time between generated frames" - so essentially the frame rate
0.05 = 20fps
fps = 1/"frame delay"
frame delay = 1/fps

**Comment by Gemcrafter** [comment]
Tested
change Scripts/Gif Creation/GifEncodeTask.cs line 166
ge.useGlobalColorTable = true;
flag from true to false

Result of 5 gif test videos.  Four produced still frames, one created a gif, with what appears to be every other frame flashing white

**Comment by Gemcrafter** [comment]
Tested adding 
ge.quality = 30;
Gif quality remains fairly low quality. 


Tested modification to
ge.quality =1;
Similar quality

files are very close to the same size. 
Video appears to be a much better solution at this point

**Comment by Gemcrafter** [comment]
VideoRecorder.cs 
contains 
        private int m_width;
        private int m_height;

Which appear to control size of the video window.  

VideoRecorderUtils.cs 
contains
            // Vertical video is disabled.
            recorder.IsPortrait = false;
            
Config file only allows changing of video resolution.
Is it possible to change the width/height or create alternate video cameras?

**Comment by Gemcrafter** [comment]
Changes in unity tested [duration][FPS](Supersampling)
Tested a complex scene
* Duration is a nice feature, might be useful to activate the circle slot (which is bad) and replace it with a higher res or longer gif

10, 1 (2) lowest framerate, visible transitions, lowest quality setting small files
5, 10 (2)  default setting (25k-50k depending on movement)
10, 30 (2)  Maxes out file size at this point at 99,968
Quality is noticeably better for most textures.  
Has noticeable difficulty with objects with full metal (high specular) 
leaves many white pixels in those places

Supersampling increase (2070 card)
10,30 (4)   Better contrast in surfaces, sharper details.  Noticeable color shift, possibly less accurate in this area.  Most textures  handled well

Probably need some mid range testing (fps) to identify when the file size cap is hit and to optimize color potentials
